src/dev/flang/be/interpreter2/LValue.java

```diff
@@ -0,0 +1,352 @@
+/*
+
+This file is part of the Fuzion language implementation.
+
+The Fuzion language implementation is free software: you can redistribute it
+and/or modify it under the terms of the GNU General Public License as published
+by the Free Software Foundation, version 3 of the License.
+
+The Fuzion language implementation is distributed in the hope that it will be
+useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+License for more details.
+
+You should have received a copy of the GNU General Public License along with The
+Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+*/
+
+/*-----------------------------------------------------------------------
+ *
+ * Tokiwa Software GmbH, Germany
+ *
+ * Source of class LValue
+ *
+ *---------------------------------------------------------------------*/
+
+package dev.flang.be.interpreter2;
+
+import dev.flang.air.Clazz;
+import dev.flang.air.Clazzes;
+
+
+/**
+ * LValue represents an address of a modifyable value type, which is the result
+ * of reading a value field.
+ *
+ * @author Fridtjof Siebert (siebert@tokiwa.software)
+ */
+public class LValue extends ValueWithClazz
+{
+
+
+  /*----------------------------  variables  ----------------------------*/
+
+
+  /**
+   * The instance (stack of heap) containing this LValue
+   */
+  public final Instance container;
+
+
+  /**
+   * The offset of this Value within container
+   */
+  public int offset;
+
+
+  /*--------------------------  constructors  ---------------------------*/
+
+
+  /**
+   * Constructor
+   *
+   * @param clazz
+   *
+   * @param outer
+   */
+  public LValue(Clazz c, Instance cont, int off)
+  {
+    super(c);
+
+    if (PRECONDITIONS) require
+      (cont != null,
+       off >= 0 || c.isUnitType(),
+       c.isUnitType() || off < Layout.get(cont._clazz).size(),
+       c.isUnitType() || off < cont.refs.length);
```

Comment:
Vertical alignment helps understanding this:
```suggestion
       c.isUnitType() || off >= 0,
       c.isUnitType() || off < Layout.get(cont._clazz).size(),
       c.isUnitType() || off < cont.refs.length);
```
Have you added the conditions or have they been remove in the main branch for some reason?

Reply:
Yes, I added these conditions